apple/internal/apple_framework_import.bzl

```diff
@@ -386,14 +506,27 @@ def _apple_static_framework_import_impl(ctx):
         transitive_sets = transitive_sets,
     ))
 
-    framework_groups = _grouped_framework_files(framework_imports)
-    framework_binaries = _all_framework_binaries(framework_groups)
+    if static_archive_imports:
+        framework_groups = {}
```

Comment:
I'm trying to use a static XCFramework created by: `apple_static_xcframework`. The filesystem structure looks like this:

```
MyModule.xcframework
├── Info.plist
├── ios-arm64
│   ├── MyModule.apple_static_library_lipo.a
│   ├── Headers
│   │   ├── MyModule.h
└── ios-arm64_x86_64-simulator
    ├── MyModule.apple_static_library_lipo.a
    ├── Headers
    │   ├── MyModule.h
```

(I manually added in a modulemap for now)

In the method: `_get_xcframework_imports` there is some code that first checks for `.framework` directories, and then checks for the `.a` without the `.framework` directories (which is this case).

Without the `.framework` directories, `framework_groups` is an empty dictionary. Line 579 `_swift_interop_info_with_dependencies` passes in this dictionary and tries to determine the `framework_name` to pass in for the modulemap support:

```
    framework_dir = framework_groups.keys()[0]
    framework_name = paths.split_extension(paths.basename(framework_dir))[0]
```

but... `framework_groups` is an empty dictionary so now crashes with: 

```
Error: index out of range (index is 0, but sequence has 0 elements)
```

How can we determine the `framework_name` in this case? Maybe the `framework_name` can be passed into this method instead of relying on `framework_groups` which could be empty?

Reply:
I think the `framework_name` can be the xcframework directory name without the `.xcframework` extension.

Reply:
I'll fix and add a test for this case.

Reply:
@codeman9 Can you test again with the latest?

Reply:
Seems to be working now. I like the separation of framework vs xcframework methods. Very easy to follow.

Reply:
No, it isn't.